src/main/java/org/broadinstitute/consent/http/db/mapper/DatasetReducer.java

```diff
@@ -37,6 +37,9 @@ public void accumulate(Map<Integer, Dataset> map, RowView rowView) {
       Integer dsIdInUse = rowView.getColumn("in_use", Integer.class);
       dataset.setDeletable(Objects.isNull(dsIdInUse));
     }
+    if (hasColumn(rowView, "dac_approval", Boolean.class)) {
+      dataset.setDacApproval(rowView.getColumn("dac_approval", Boolean.class));
```

Comment:
I'm not a complete expert on Mappers and Reducers, but my understanding was that Reducers (specifically the ones we use that extend `LinkedHasMapRowReducer`) operate similarly to Mappers with one exception: it stores the row result as a key-value pair, with the key being a user designated identifier (usually an id). 

The difference kicks in when a model's value is repeated on multiple rows (usually on a JOIN). Both mappers and reducers will tackle their operations row by row, but a mapper is naive and will just create an instance and pass it to the resultant list. A reducer will check to see if the model is already saved in the map. If it's there it can just fetch the currently stored value and update it, otherwise it'll add it to the array. The end result is that the mapper will have multiple instances of a model with the same id, whereas the reducer would not. That said, a reducer is going to be slower since it has to do things in the background (one being tracking insertion order).

The bug is a concern though, I keep forgetting that Java has both `boolean` and `Boolean`. I think a cast should solve the issue, but I'll add tests to check on the null values just to be safe.

Reply:
Makes sense, thanks for the explanation! Do you have any tips on determining if a particular column name should be checked in the mapper/reducer/both?

Reply:
I like to do column checks in both mappers and reducers. One more thing to remember, Reducers and mappers also go hand in hand. When a reducer tries to instantiate an object from a row, it will start with the existing mapper. We usually go a step further to populate additional info depending on the various logic conditions.

Reply:
So for #1709 I only added 
```
if (hasColumn(r, "sharing_plan_document")) {
      dataset.setSharingPlanDocument(r.getString("sharing_plan_document"));
 }
```
in `DatasetMapper` and not `DatasetReducer` since the reducer calls the mapper to instantiate the object. Is this correct, or is additional code needed in the reducer?

Reply:
If the mapper reads it, then that will be enough to instantiate the field. Sometimes in the reducer, we need to do extra work to populate fields, and that's where I would want to double check that the columns I wanted existed.